src/services/bookmarks/bookmarks.service.js

```diff
@@ -10,38 +10,55 @@
 angular.module('vlui')
   .service('Bookmarks', function(_, vl, localStorageService, Logger, Dataset) {
     var Bookmarks = function() {
-      this.dict = {};
+      this.dict = [];
       this.length = 0;
       this.isSupported = localStorageService.isSupported;
+
+      this.clearDeprecatedBookmarks();
     };
 
     var proto = Bookmarks.prototype;
 
     proto.updateLength = function() {
-      this.length = Object.keys(this.dict).length;
+      this.length = this.dict.length;
     };
 
     proto.save = function() {
       localStorageService.set('bookmarks', this.dict);
     };
 
     proto.load = function() {
-      this.dict = localStorageService.get('bookmarks') || {};
+      this.dict = localStorageService.get('bookmarks') || [];
       this.updateLength();
     };
 
+    /**
+     * Clear deprecated bookmarks in localStorage. 
+     * We switched from this.dict = {} to this.dict = [],
+     * this.dict = {} may be still in localStorage and cause problems on this.load() 
+     * This function is mostly beneficial to developers working on this repo.
+     */
+    proto.clearDeprecatedBookmarks = function() {
```

Comment:
I wonder if keeping code like this is a good idea in longer term?  

Could we just use different names (e.g., `bookmarkList` ) and get rid of this entirely